### PR TITLE
Add Spot Scaling error message

### PR DESCRIPTION
**Pull request by azure562**
- Add an FAQ entry about an error message
- Group FAQ entries

### CODE REVIEW

File: website/docs/spot-scaling/faq.mdx [hunk 1]
```diff
@@ -4,15 +4,17 @@ sidebar_position: 3
...
 # Spot Scaling FAQ
 
-## What is a Spot Instance?
+## Spot Instance
 
-Spot Instances are spare EC2 capacity that can save you up to 90% off On-Demand prices that Amazon Web Services can interrupt with a two-minute notification. Spot uses the same underlying EC2 instances as On-Demand and Reserved Instances and is best suited for fault-tolerant, flexible workloads. In addition, spot instances provide an additional option for obtaining compute capacity and can be used along with On-Demand and Reserved Instances.
+### What is a Spot Instance?
 
-## What are the differences between Spot Instances and On-Demand Instances?
+Spot Instances are spare EC2 capacity that can save you up to 90% off On-Demand prices that Amazon Web Services can interrupt with a two-minute notification. Spot uses the same underlying EC2 instances as On-Demand and Reserved Instances and is best suited for fault-tolerant, flexible workloads. Spot instances provide an additional option for obtaining compute capacity and can be used along with On-Demand and Reserved Instances.
```
Comment (norosa) [inline website/docs/spot-scaling/faq.mdx:11]: Do you think it makes sense to hyperlink some of the noun phrases to the AWS docs? (Or maybe we could add a SEE ALSO admonition to this paragraph.)

Nit: Is it proper usage to refer to them as "Spot" without "Instances"?

File: website/docs/spot-scaling/faq.mdx [hunk 2]
```diff
@@ -4,15 +4,17 @@ sidebar_position: 3
...
 
-While running, Spot Instances are essentially the same as On-Demand Instances. The main differences are that Spot Instances typically offer a significant discount over On-Demand prices. The tradeoff is that your Spot Instances can be interrupted by Amazon EC2 for capacity requirements with a two-minute notification. Spot prices adjust gradually based on long-term supply and demand for spare EC2 capacity. See [Amazon EC2 Spot Instances](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-spot-instances.html) for more information.
+### What are the differences between Spot Instances and On-Demand Instances?
 
-## When to use Spot Instances?
+While running, Spot Instances are essentially the same as On-Demand Instances. The main difference is that Spot Instances typically offer a significant discount over On-Demand prices. The tradeoff is that Spot Instances can be interrupted by Amazon EC2 for capacity requirements with a two-minute notification. Spot prices adjust gradually based on long-term supply and demand for spare EC2 capacity. See [Amazon EC2 Spot Instances](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-spot-instances.html) for more information.
```
Comment (norosa) [inline website/docs/spot-scaling/faq.mdx:15]: We could pull the link to the AWS docs into a SEE ALSO admonition here too, to match the previous section (depending on what you want to do with that section).

You can create a SEE ALSO admonition like this:

```
::: See also

Admonition text here.

:::
```

**Review by norosa — COMMENTED** [review]
Thanks for the PR! Left some comments.

BTW, did you mean to replace the `website/assets/png/cmp-spot-scaling-not-optimized.png` file? I noticed that the new version of the file uses standard resolution instead of Retina (@2x). Was that intentional?

//Sorry for replying here, the image size issue was not by intention, it's just I didn't notice it

File: website/docs/spot-scaling/faq.mdx (hunk 2 above)
Comment (azure562) [reply website/docs/spot-scaling/faq.mdx:15]: Let's put such topics in a style guide and then do n overhaul

File: website/docs/spot-scaling/faq.mdx [hunk 3]
```diff
@@ -22,52 +24,73 @@ Additional use cases include:
...
 
 Spot Scaling Auto-Enrollment behaves the same as when you apply the recommendations in the CMP. By default Spot Scaling recommends 20% of desired capacity exists On-Demand and 80% with Spot Instances. The same for the allowed instance type, Spot Scaling recommends similar machine types based on vCPUs and memory configuration.
+
+## Error messages
+
+### Incompatible launch template
+
+This error happens when you try to create a mixed instances group using an unsupported launch template version.
+
+To benefit from all Spot Scaling features, you need to create a launch template _without_ specifying the **Purchasing option** of `Request Spot Instances` in the **Advanced Details** section.
```
Comment (fdelacortina) [inline website/docs/spot-scaling/faq.mdx:87]: I will say:

To benefit from all Spot Scaling features, you need to create a launch template or a new version of the same _without_ specifying the **Purchasing option** of `Request Spot Instances` in the **Advanced Details** section.

File: website/docs/spot-scaling/faq.mdx [hunk 4]
```diff
@@ -22,52 +24,73 @@ Additional use cases include:
...
+
+<picture
+  img={require('../../assets/png/cmp-spot-instance-purchase.png')}
+  alt="A screenshot showing AWS Launch Template purchasing option"
+/>
+
+A launch template lets you configure additional settings in your Auto Scaling group to launch multiple instance types and combine On-Demand and Spot purchase options. To launch instances with such a combination, you must _not_ specify Spot Instance request in the launch template (see [Limitations](https://docs.aws.amazon.com/autoscaling/ec2/userguide/create-launch-template.html#create-launch-template-limitations)). Otherwise, you would see the following error message on the CMP ASGs screen: `Spotisize error: incompatible launch template: already requesting for spot instances`.
```
Comment (fdelacortina) [inline website/docs/spot-scaling/faq.mdx:94]: The final point (.) should be included in the error message or outside text? In below text (Line 96) your are including it into the error message.

File: website/docs/spot-scaling/faq.mdx (hunk 3 above)
Comment (azure562) [reply website/docs/spot-scaling/faq.mdx:87]: changed to "create a new launch template or a new version of the same template"

File: website/docs/spot-scaling/faq.mdx (hunk 4 above)
Comment (azure562) [reply website/docs/spot-scaling/faq.mdx:94]: The error msg from AWS is a full sentence, so I guess the period is part of the msg. The CMP msg doesn't have that period in itself. To be consistent, I removed the one here

File: website/docs/spot-scaling/faq.mdx [hunk 5]
```diff
@@ -22,52 +24,73 @@ Additional use cases include:
...
-## What is the default behavior of Spot Scaling Auto-Enrollment?
+### What is the default behavior of Spot Scaling Auto-Enrollment?
 
 Spot Scaling Auto-Enrollment behaves the same as when you apply the recommendations in the CMP. By default Spot Scaling recommends 20% of desired capacity exists On-Demand and 80% with Spot Instances. The same for the allowed instance type, Spot Scaling recommends similar machine types based on vCPUs and memory configuration.
+
+## Error messages
+
+### Incompatible launch template
+
+This error happens when you try to create a mixed instances group using an unsupported launch template version.
```
Comment (fdelacortina) [inline website/docs/spot-scaling/faq.mdx:85]: Please use following sentence here: "This error happens when Spot Scaling tries to create a mixed instances group using a launch template that already requests Spot Instances."

File: website/docs/spot-scaling/faq.mdx [hunk 6]
```diff
@@ -22,52 +24,73 @@ Additional use cases include:
...
+
+To benefit from all Spot Scaling features, you need to create a new launch template or a new version of the same template _without_ specifying the **Purchasing option** of `Request Spot Instances` in the **Advanced Details** section.
+
+<picture
+  img={require('../../assets/png/cmp-spot-instance-purchase.png')}
+  alt="A screenshot showing AWS Launch Template purchasing option"
+/>
+
+A launch template lets you configure additional settings in your Auto Scaling group to launch multiple instance types and combine On-Demand and Spot purchase options. To launch instances with such a combination, you must _not_ specify Spot Instance request in the launch template (see [Limitations](https://docs.aws.amazon.com/autoscaling/ec2/userguide/create-launch-template.html#create-launch-template-limitations)). Otherwise, you would see the following error message on the CMP ASGs screen: `Spotisize error: incompatible launch template: already requesting for spot instances`
```
Comment (fdelacortina) [inline website/docs/spot-scaling/faq.mdx:94]: Please remove "Spotize error:" and leave it like this:  ' Incompatible launch template: already requesting for spot instances`

File: website/docs/spot-scaling/faq.mdx [hunk 7]
```diff
@@ -22,52 +24,73 @@ Additional use cases include:
...
 Spot Scaling Auto-Enrollment is tool-agnostic in the sense that the tags you need (`doitintl:spotscaling:auto-enroll` and `doitintl:fallback-to-on-demand`) are independent of the IaC tool you use.
 
-## What is the default behavior of Spot Scaling Auto-Enrollment?
+### What is the default behavior of Spot Scaling Auto-Enrollment?
 
 Spot Scaling Auto-Enrollment behaves the same as when you apply the recommendations in the CMP. By default Spot Scaling recommends 20% of desired capacity exists On-Demand and 80% with Spot Instances. The same for the allowed instance type, Spot Scaling recommends similar machine types based on vCPUs and memory configuration.
+
+## Error messages
+
+### Incompatible launch template
```
[the remark this answers is not in the document]
Comment (azure562) [reply website/docs/spot-scaling/faq.mdx:83]: if we change it, the URL will be different. Do you still want to make the change?

File: website/docs/spot-scaling/faq.mdx [hunk 8]
```diff
@@ -22,52 +24,73 @@ Additional use cases include:
...
+
+This error happens when you try to create a mixed instances group using an unsupported launch template version.
+
+To benefit from all Spot Scaling features, you need to create a new launch template or a new version of the same template _without_ specifying the **Purchasing option** of `Request Spot Instances` in the **Advanced Details** section.
```
Comment (fdelacortina) [inline website/docs/spot-scaling/faq.mdx:87]: Remove 'To benefit from all Spot Scaling features' and change it for ' and change it for 'To solve'.

**Review by fdelacortina — APPROVED** [review]

**Pull request merged**